src/components/PopupWithForm.js

```diff
@@ -1,33 +1,71 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit, formValidator) {
+  constructor(popupSelector, handleFormSubmit, formValidator = null) {
     super(popupSelector);
-    this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popup.querySelector(".modal__form");
-    this._formValidator = formValidator;
+    this.handleFormSubmit = handleFormSubmit;
+    this.form = this._popup.querySelector(".modal__form");
+    this.formValidator = formValidator;
+    this.submitButton = this.form.querySelector(".modal__button");
+    this.submitButtonText = this.submitButton.textContent;
   }
 
-  _getInputValues() {
-    this._formValues = {};
-    this._form.querySelectorAll(".modal__input").forEach((input) => {
-      this._formValues[input.name] = input.value;
+  getInputValues() {
+    const formValues = {};
+    this.form.querySelectorAll(".modal__input").forEach((input) => {
+      formValues[input.name] = input.value;
+    });
+    return formValues;
+  }
+
+  setInputValues(data) {
+    this.form.querySelectorAll(".modal__input").forEach((input) => {
+      input.value = data[input.name];
     });
-    return this._formValues;
   }
 
   setEventListeners() {
-    super.setEventListeners();
-    this._form.addEventListener("submit", (event) => {
+    this.form.addEventListener("submit", (event) => {
       event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._form.reset();
-      this._formValidator.toggleButtonState();
+      this.renderLoading(true); // Show "Saving..." text
+      this.handleFormSubmit(this.getInputValues())
+        .then(() => {
+          this.form.reset();
+          if (this.formValidator) {
+            this.formValidator.toggleButtonState();
+          }
+          this.close();
+        })
+        .catch((err) => {
+          console.error("Form submission error:", err);
+        })
+        .finally(() => {
+          this.renderLoading(false);
+        });
     });
+    super.setEventListeners();
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this.submitButton.textContent = "Saving...";
+    } else {
+      this.submitButton.textContent = this.submitButtonText;
+    }
   }
 
   open() {
     super.open();
-    this._formValidator.resetValidation();
+    if (this.formValidator) {
+      this.formValidator.resetValidation();
+    }
+  }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.